hopli/src/methods.rs

```diff
@@ -0,0 +1,2226 @@
+
+    for node in node_addresses {
+        multicall.add_call(
+            network_registry
+                .method::<_, Address>("nodeRegisterdToAccount", node)
+                .map_err(|e| MulticallError::ContractError(ContractError::AbiError(e)))?,
+            false,
+        );
+    }
+
+    let response: Vec<Address> = multicall.call_array().await?;
+
+    Ok(response)
+}
+
+/// Register safes and nodes to the network registry, and force-sync the eligibility to true.
+/// It returns the number of removed nodes and nodes being added.
+/// - If nodes have been registered to a different safe, overwrite it (remove the old safe and regsiter with the new safe)
+/// - If ndoes have been registered to the same safe, no op
+/// - If nodes have not been registered to any safe, register it
+/// After all the nodes have been added to the network registry, force-sync the eligibility of all the added safes to true
+pub async fn register_safes_and_nodes_on_network_registry<M: Middleware>(
+    network_registry: HoprNetworkRegistry<M>,
+    safe_addresses: Vec<H160>,
+    node_addresses: Vec<H160>,
+) -> Result<(usize, usize), HelperErrors> {
+    assert_eq!(
+        safe_addresses.len(),
+        node_addresses.len(),
+        "unmatched lengths of safes and nodes"
+    );
+
+    // check registered safes of given node addresses
+    let registered_safes =
+        get_registered_safes_for_nodes_on_network_registry(network_registry.clone(), node_addresses.clone())
+            .await
+            .unwrap();
+
+    let mut nodes_to_remove: Vec<H160> = Vec::new();
+    let mut safes_to_add: Vec<H160> = Vec::new();
+    let mut nodes_to_add: Vec<H160> = Vec::new();
+
+    for (i, registered_safe) in registered_safes.iter().enumerate() {
+        if registered_safe.eq(&H160::zero()) {
+            // no entry, add to network registry
+            safes_to_add.push(safe_addresses[i]);
+            nodes_to_add.push(node_addresses[i]);
+        } else if registered_safe.ne(&safe_addresses[i]) {
+            // remove first then add
+            nodes_to_remove.push(node_addresses[i]);
+            safes_to_add.push(safe_addresses[i]);
+            nodes_to_add.push(node_addresses[i]);
+        } else {
+            // no-op
+        }
+    }
+
+    if !nodes_to_remove.is_empty() {
+        // need to remove some nodes
+        network_registry
+            .manager_deregister(nodes_to_remove.clone())
+            .send()
+            .await
+            .unwrap()
+            .await
+            .unwrap();
+    }
+
+    network_registry
+        .manager_register(safes_to_add.clone(), nodes_to_add.clone())
+        .send()
+        .await
+        .unwrap()
+        .await
+        .unwrap();
+
+    // force sync their eligibility
+    network_registry
+        .manager_force_sync(safes_to_add.clone(), vec![true; safes_to_add.len()])
+        .send()
+        .await
+        .unwrap()
+        .await
+        .unwrap();
```

Comment:
Shouldn't do `unwrap` in a function that already returns `Result`, right ?

Reply:
Should not do `unwrap()` at all, unless a previous `is_ok()` check is present.